src/sentry/search/events/datasets/functions.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Mapping, Optional, Union
+
+from snuba_sdk import Column as SnColumn
+from snuba_sdk import Condition, Direction, Op, OrderBy
+
+from sentry.api.event_search import SearchFilter
+from sentry.search.events import builder
+from sentry.search.events.constants import EQUALITY_OPERATORS, PROJECT_ALIAS, PROJECT_NAME_ALIAS
+from sentry.search.events.datasets import field_aliases, filter_aliases
+from sentry.search.events.datasets.base import DatasetConfig
+from sentry.search.events.fields import (
+    ColumnArg,
+    Combinator,
+    Function,
+    InvalidFunctionArgument,
+    InvalidSearchQuery,
+    NumberRange,
+    NumericColumn,
+    SnQLFunction,
+    with_default,
+)
+from sentry.search.events.types import NormalizedArg, ParamsType, SelectType, WhereType
+
+
+class Kind(Enum):
+    DATE = "date"
+    DURATION = "duration"
+    INTEGER = "integer"
+    NUMBER = "number"
+    STRING = "string"
+
+
+class Duration(Enum):
+    NANOSECOND = "nanosecond"
+    MICROSECOND = "microsecond"
+    MILLISECOND = "millisecond"
+    SECOND = "second"
+    MINUTE = "minute"
+    HOUR = "hour"
+    DAY = "day"
+    WEEK = "week"
+
+
+# The only units available right now are duration based
+Unit = Duration
+
+
+@dataclass(frozen=True)
+class Column:
+
+    # the internal name in snuba
+    column: str
+    # type kind/type associated with this column
+    kind: Kind
+    # the external name to expose
+    alias: Optional[str] = None
+    # some kinds will have an unit associated with it
+    unit: Optional[Unit] = None
```

Comment:
I wonder if you can just import a lot of this boilerplate from the profiles dataset?

Reply:
yeah with some changes we can.. the Function/Numeric cols reference a module scoped variable `COLUMN_MAP` etc. ideally since its all effectively the same we modify it and move it to a common folder.

another change i'd like to handle after